xgcm/metadata_parser.py

```diff
@@ -0,0 +1,51 @@
+import warnings
+
+import cf_xarray
+
+
+def cf_parser(ds):
```

Comment:
We should type hint this, probably with [TypedDict](https://peps.python.org/pep-0589/)